substrate/client/service/src/builder.rs

```diff
@@ -507,8 +508,21 @@ where
 		)
 	};
 
-	let rpc = start_rpc_servers(&config, gen_rpc_module, rpc_id_provider)?;
-	let rpc_handlers = RpcHandlers(Arc::new(gen_rpc_module(sc_rpc::DenyUnsafe::No)?.into()));
+	let (rpc, listen_addr) = start_rpc_servers(&config, gen_rpc_module, rpc_id_provider)?;
+
+	let listen_addrs = match listen_addr {
+		Some(socket_addr) => {
+			let mut multiaddr: Multiaddr = socket_addr.ip().into();
+			multiaddr.push(Protocol::Tcp(socket_addr.port()));
+			vec![multiaddr]
+		},
+		None => vec![],
+	};
+
+	let rpc_handlers = RpcHandlers {
```

Comment:
It doesn't makes sense to add listen_addresses to `RpcHandlers` this just an in-memory way to perform rpc calls, please remove it RpcHandlers.

Why did you add it (maybe I'm missing something)?

Reply:
>Why did you add it (maybe I'm missing something)?

Just because the RpcHandlers type is what is getting returned to the caller. Would a new return type that can provide both the ```Server``` and the ```RpcHandlers``` be preferred?

Reply:
Aight, I see one may want to know the listen addrs on the rpc endpoint after calling `spawn_tasks`.

It's probably fine to keep the `RpcHandlers` or rename to the RpcService or something, but it needs to documented that is represent a running RPC server as well with the legacy `in-memory rpc calls`.

We may want to remove latter at some point, it was for WASM stuff before smoldot was a thing...

Reply:
I added some notes with a link to this comment thread.